src/main/java/practice/StreamPractice.java

```diff
@@ -59,7 +81,13 @@ public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
      * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
      */
     public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(p -> (!p.getCats().isEmpty() && p.getSex().equals(Person.Sex.WOMAN)
```

Comment:
we don't need this check
```suggestion
                .filter(p -> p.getSex().equals(Person.Sex.WOMAN)
```

Reply:
why ? does not indicate that the "Cats" field is not empty.